Source/player.cpp

```diff
@@ -1139,17 +1139,17 @@ bool PlrDirOK(int pnum, Direction dir)
 	auto &player = plr[pnum];
 
 	Point position = player.position.tile + dir;
-	if (position.x < 0 || !dPiece[position.x][position.y] || !PosOkPlayer(pnum, position.x, position.y)) {
+	if (position.x < 0 || !dPiece[position.x][position.y] || !PosOkPlayer(pnum, position)) {
 		return false;
 	}
 
 	isOk = true;
 	if (dir == DIR_E) {
-		isOk = !SolidLoc(position.x, position.y + 1) && !(dFlags[position.x][position.y + 1] & BFLAG_PLAYERLR);
+		isOk = !SolidLoc(position + Point { 0, 1 }) && !(dFlags[position.x][position.y + 1] & BFLAG_PLAYERLR);
 	}
 
 	if (isOk && dir == DIR_W) {
-		isOk = !SolidLoc(position.x + 1, position.y) && !(dFlags[position.x + 1][position.y] & BFLAG_PLAYERLR);
+		isOk = !SolidLoc(position + Point { 1, 0 }) && !(dFlags[position.x + 1][position.y] & BFLAG_PLAYERLR);

```

Comment:
```suggestion
		isOk = !SolidLoc(position + DIR_SW) && !(dFlags[position.x][position.y + 1] & BFLAG_PLAYERLR);
```

Reply:
Not sure why you marked this as resolved but didn't apply it, but applied the exact same change to the monster version of this function.

Reply:
That was me having a brain fart. I thought the comments were duplicated. Fixed now.